shuffle/cli/src/main.rs

```diff
@@ -133,3 +149,19 @@ fn normalized_key_path(diem_root_key_path: Option<PathBuf>) -> Result<PathBuf> {
         }
     }
 }
+
+fn normalized_network(network: &str) -> Result<Url> {
+    let net = match Url::parse(network) {
+        Ok(_res) => Url::parse(network),
+        Err(_e) => Url::parse(("http://".to_owned() + network).as_str()),
+    };
+    Ok(net?)
```

Comment:
L154 and L155 are parsing twice unnecessarily. You can collapse this match block further by returning directly:

```rust
    match Url::parse(network) {
        Ok(res) => return Ok(res),
        Err(_e) => return Ok(Url::parse(("http://".to_owned() + network).as_str())?),
    };
```